source/server/listener_manager_impl.cc

```diff
@@ -401,7 +401,7 @@ void ListenerManagerImpl::drainListener(ListenerImplPtr&& listener) {
   // Tell all workers to stop accepting new connections on this listener.
   draining_it->listener_->debugLog("draining listener");
   for (const auto& worker : workers_) {
-    worker->stopListener(*draining_it->listener_);
+    worker->stopListener(*draining_it->listener_, []() {});
```

Comment:
nit: probably pass `nullptr` instead of empty callback and call the callback if it is not null in `stopListener` method - might be cleaner?